packages/protocol/migrations_sol/Migration.s.sol

```diff
@@ -404,7 +426,32 @@ contract Migration is Script, UsingRegistry {
     IReserve(registry.getAddressForStringOrDie("Reserve")).addToken(stableTokenProxyAddress);
 
     getFeeCurrencyWhitelist().addToken(stableTokenProxyAddress);
+    /* 
+    TODO(Arthur): Add stabletoken to FeeCurrencyDirectory 
+    1. Check that the syntax below is correct to .setCurrencyCongif
+    2. Set oracle address (is this sortedOracle?)
+    3. Set intrinsic gas value
+    */
+    /*
+    This is an arbitrary hex address I took from celoscan.
+    For now, I only want to test if I can set a currency config.
+    */
+    MockOracle mockOracle = new MockOracle();
```

Comment:
Can we use SortedOracles instead ?

Reply:
Yes, it looks like the latest [`SortedOracles.sol`](https://github.com/celo-org/celo-monorepo/blob/a4c608b4211794bb1651386785d2661b9679e8b7/packages/protocol/contracts/stability/SortedOracles.sol) implementation has a `getExchangeRate()` implementation:

https://github.com/celo-org/celo-monorepo/blob/ed539fda163b1f5d7ce6a2b40944891c4c3a9ad2/packages/protocol/contracts/stability/SortedOracles.sol#L329-L333

Last I looked, [`SortedOracles.sol`](https://github.com/celo-org/celo-monorepo/blob/a4c608b4211794bb1651386785d2661b9679e8b7/packages/protocol/contracts/stability/SortedOracles.sol) did not implement the [`IOracle.sol`](https://github.com/celo-org/celo-monorepo/blob/43a3146c068de25cda53b0aff4853ea4bc37e887/packages/protocol/contracts-0.8/common/interfaces/IOracle.sol) interface. 

Nit: Looks like the class definition does not specify that is "`is IOracle`". Any reason we didn't add this? Happy to make this small change if it makes sense.

https://github.com/celo-org/celo-monorepo/blob/ed539fda163b1f5d7ce6a2b40944891c4c3a9ad2/packages/protocol/contracts/stability/SortedOracles.sol#L39-L39

Reply:
This is a miss. If you can, please add IOracle to SortedOracles

Reply:
https://github.com/celo-org/celo-monorepo/pull/10992/commits/78e6cf273e87f2445bed4912f99486bce93f5b08 adds `IOracle` to the class definition

Reply:
> As of [1c04e8b](https://github.com/celo-org/celo-monorepo/commit/1c04e8ba7ea1c632754709bd560f7b9c58f0aa90), the migration script sets the oracle in `CurrencyConfig` to the [`SortedOracles.sol`](https://github.com/celo-org/celo-monorepo/blob/a4c608b4211794bb1651386785d2661b9679e8b7/packages/protocol/contracts/stability/SortedOracles.sol) contract (which implements [IOracle.sol](https://github.com/celo-org/celo-monorepo/blob/43a3146c068de25cda53b0aff4853ea4bc37e887/packages/protocol/contracts-0.8/common/interfaces/IOracle.sol)).
> 
> This works as expected ✅
> 
> Calling the `getExchangeRate` function on a registered currency returns an exchange rate:
> 
> ```shell
> $ cast call \
> 0x42Fe5a2A61ed9705eb2F08a04A58CEB606D22f6a \
> "getExchangeRate(address)(uint256, uint256)" \
> 0xb7a33b4ad2B1f6b0a944232F5c71798d27Ad9272 \
> --rpc-url=http://127.0.0.1:8546
> 1000000000000000000000000 [1e24]
> 1000000000000000000000000 [1e24]
> ```

Source: https://github.com/celo-org/celo-monorepo/pull/10992#issuecomment-2120486484